sdk/data/azcosmos/cosmos_global_endpoint_manager.go

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package azcosmos
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+
+	azruntime "github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+)
+
+const defaultBackgroundRefreshInterval = 5 * time.Minute
+
+type globalEndpointManager struct {
+	client           *Client
+	preferredRegions []string
+	locationCache    *locationCache
+}
+
+// newGlobalEndpointManager creates a new global endpoint manager.
+// It takes a client, preferredRegions, and refreshInterval as input parameters.
+// If the refreshInterval is zero, it uses the default value (5 minutes).
+// It initializes a new globalEndpointManager, creates a location cache, and starts a background refresh process.
+// The background refresh process periodically updates the location cache based on the specified refreshInterval.
+func newGlobalEndpointManager(client *Client, preferredRegions []string, refreshInterval time.Duration) (*globalEndpointManager, error) {
+	endpoint, err := url.Parse(client.endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	// If refreshInterval is zero, use the default value (5 minutes)
+	if refreshInterval == 0 {
+		refreshInterval = defaultBackgroundRefreshInterval
+	}
+
+	gem, err := &globalEndpointManager{
+		client:           client,
+		preferredRegions: preferredRegions,
+		locationCache:    newLocationCache(preferredRegions, *endpoint),
+	}, nil
+	if err != nil {
```

Comment:
How is `err != nil` if it's being set to `nil` on the previous line? Should this line be removed?